src/store/filestore.rs

```diff
@@ -89,17 +100,31 @@ impl BlobInterface for LocalFileStore {
     }
 }
 
+#[expect(clippy::unwrap_used, reason = "Valid static regex")]
+static RE_MODEL_METADATA: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(
+        r"(?x)
+            ^
+                (?<store_directory>.*)\/
+                    (?<namespace>[a-z_]+)\/
+                        (?<class>[a-z_]+)\/
+                            (?<hash>[0-9a-f]+)\/
+                                (
+                                    annotation\/
+                                        (?<name>[0-9a-zA-Z\-]+)
+                                        -
+                                        (?<version>[0-9]+\.[0-9]+\.[0-9]+)
+                                        \.yaml
+                                |
+                                    spec\.yaml
+                                )
+            $
+            ",
+    )
+    .unwrap()
+});
+
 impl LocalFileStore {
-    /// Get the directory where store is located.
```

Comment:
Agreee with the removal of this, since I removed it too initially before the last pull request